### PR TITLE
Update to latest kinvey-js-sdk

### PR DESCRIPTION
**Pull request by thomasconner**
### Description
Updates the Kinvey BackboneJS SDK to use the latest kinvey-js-sdk.

### CODE REVIEW

**Comment by thomasconner** [comment]
Added https://github.com/KinveyApps/Backbone-Starter

File: README.md [hunk 1]
```diff
@@ -82,6 +83,8 @@ The steps for running the end to end tests is as follows:
 3. Execute `npm run e2e:server`. __Keep this terminal window open.__
 
 #### Start App
+__Requirement:__ Execute `npm install -g yo generator-kinvey-html` to globally install [Yeoman](http://yeoman.io/) and [generator-kinvey-html](https://www.npmjs.com/package/generator-kinvey-html). You only need to do this once.
```
Comment (tejasranade) [inline README.md]: @thomasconner does this apply to backbone?

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "kinvey-backbone-sdk",
   "version": "3.2.2",
```
Comment (tejasranade) [inline package.json]: @thomasconner I'm assuming this will be updated before you release. It will jump to 3.4.1, right?

File: package.json [hunk 3]
```diff
@@ -41,30 +42,25 @@
...
-    "lodash": "^4.16.2"
-  },
-  "peerDependencies": {
-    "backbone": "^1.3.3",
-    "core-js": "^2.4.1",
-    "es6-promise": "^4.0.3",
-    "kinvey-phonegap-sdk": "3.2.2",
-    "kinvey-node-sdk": "3.2.2",
-    "lodash": "^4.16.2"
+    "kinvey-node-sdk": "3.4.0",
```
Comment (tejasranade) [inline package.json]: @thomasconner are you sure this is correct? Shouldn't one of these two (node or phonegap) be sufficient? In fact, does phonegap make sense here given that most backbone apps will run in the web browser?

File: webpack.config.js [hunk 4]
```diff
@@ -32,6 +32,15 @@ module.exports = {
     path: path.resolve(__dirname, 'dist')
   },
   plugins: [
-    new webpack.BannerPlugin(BANNER, { raw: true })
-  ]
+    new webpack.BannerPlugin(BANNER, { raw: true }),
+    new webpack.NormalModuleReplacementPlugin(
+      /kinvey-node-sdk\/dist\/identity\/src\/popup\.js/,
+      require.resolve(path.resolve(__dirname, 'node_modules/kinvey-phonegap-sdk/dist/popup.js'))
```
Comment (tejasranade) [inline webpack.config.js]: @thomasconner again, `phonegap` seems a little odd here.

File: README.md (hunk 1 above)
Comment (thomasconner) [inline README.md]: Oops. Nope.

File: package.json (hunk 2 above)
Comment (thomasconner) [inline package.json]: Yes

File: package.json (hunk 3 above)
Comment (thomasconner) [inline package.json]: Most backbone apps will run in the web browser but isn't it possible someone could build a PhoneGap app with Backbone?

File: webpack.config.js (hunk 4 above)
Comment (thomasconner) [inline webpack.config.js]: Same as above

**Pull request closed without merging**